stdlib/map.mli

```diff
@@ -114,17 +114,17 @@ module type S =
 
     val merge:
          (key -> 'a option -> 'b option -> 'c option) -> 'a t -> 'b t -> 'c t
-    (** [merge f m1 m2] computes a map whose keys is a subset of keys of [m1]
+    (** [merge f m1 m2] computes a map whose keys are a subset of the keys of [m1]
         and of [m2]. The presence of each such binding, and the corresponding
         value, is determined with the function [f].
         In terms of the [find_opt] operation, we have
-        [find_opt x (merge f m1 m2) = f (find_opt x m1) (find_opt x m2)]
-        for any key [x], provided that [f None None = None].
+        [find_opt x (merge f m1 m2) = f x (find_opt x m1) (find_opt x m2)]
+        for any key [x], provided that [f x None None = None].
         @since 3.12.0
      *)
 
     val union: (key -> 'a -> 'a -> 'a option) -> 'a t -> 'a t -> 'a t
-    (** [union f m1 m2] computes a map whose keys is the union of keys
+    (** [union f m1 m2] computes a map whose keys are a subset of the keys
```

Comment:
This new sentence feels a bit imprecise since it left out the lower bound in 
```
keys(m1) ∆ keys(m2) ⊂ keys(union g m1 m2) ⊂ keys(m1) ∪ keys(m2)
```
Maybe we should amend the previous sentence to:

> [union f m1 m2] computes a map whose keys is the union of keys minus the keys `k` for which
      [f k (find k m1) (find k m2) = None].

?

Reply:
I would just use `a subset of the union of the keys of [m1] and [m2]` -- the formulation proposed by Jeremy is maybe not explicit enough in the fact that "subset of the keys of [m1] and of [m2]" should be interpreted disjunctively, not conjunctively. The role of `f` should be clarified by the rest of the documentation (which could maybe be clearer than it is today, but this is orthogonal).

Reply:
One reason that the documentation needs to be either imprecise or complex is because the behaviour of the function is quite complex (because of the `option` in the return type of `f`).

[As originally proposed](https://github.com/ocaml/ocaml/issues/6449), the `union` function had a simpler type:

```ocaml
val union: (key -> 'a -> 'a -> 'a) -> 'a t -> 'a t -> 'a t
```

With this type the existing simple documentation (`union of keys`) is correct, which suggests that the function is easier to understand.  I wonder why it was changed.  (@alainfrisch, do you remember?)

Reply:
I'm going to propose to leave this part of the documentation as it is, even though it could be more precise (at the cost of simplicity).

The current description has a definite bug, which this PR directly fixes; others are always free to make further improvements as independent changes later on.

Reply:
> (@alainfrisch, do you remember?)

I can only reconstruct it as posteriori, but the rationale was probably that the version returning an `option` doesn't make the function more complex (both for runtime and code complexity), but strictly more useful; and we wouldn't want to add yet a third similar function to Map.